pkg/performance/registry.go

```diff
@@ -28,6 +48,85 @@ func Register(metricType MetricType, collector NewContinuousCollector) {
 	registry[metricType] = collector
 }
 
+// TryRegister attempts to register a collector after checking if it can run on the current platform.
+// If the collector cannot run due to missing capabilities or incompatible platform, it is tracked
+// in the unavailable collectors list with the reason.
+//
+// This function is called during package initialization and will not panic on capability failures.
+func TryRegister(metricType MetricType, collector NewContinuousCollector) {
```

Comment:
I would think we both `TryRegister` and `Register` should check for capabilities. But `Register` would be a hard check so will panic if the process doesn't have the required capabilities while `TryRegister` is a soft check would will just print an warning say it can't register it